### PR TITLE
api/types: move various types to api/types/(images|containers|swarm)  

### PR DESCRIPTION
**Pull request by thaJeztah**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
Looks like I never pushed this one (I have another one pending, but that one still needed some work around search and auth to prevent a circular import)

**Review by vvoland — APPROVED** [review]
LGTM; although it's probably worth noting that we are starting to have too many `image`(s) modules 😄 

```go
"github.com/containerd/containerd/images"
"github.com/docker/docker/image"
"github.com/docker/docker/daemon/images"
imagetypes "github.com/docker/docker/api/types/image"
```

**Comment by thaJeztah** [comment]
> although it's probably worth noting that we starting to have too many image(s) modules 😄

OHMAN, yes, we do, and with the containerd-image-store integration that definitely became more apparent. We may be able to avoid _some_ of these if we work on cleaner separation of API types and internal types, but I guess it's something we will have to deal with for some time.

Perhaps we should enforce aliases for some of these; same as we did for the containerd-errdefs package; https://github.com/moby/moby/blob/aacd100a70ec3297e0480e9d1f2c8ba9d564bf82/hack/validate/golangci-lint.yml#L33-L36

**Comment by thaJeztah** [comment]
oh! needs another rebase (due to the containerd logs changes)

File: api/server/router/swarm/backend.go [hunk 1]
```diff
@@ -3,40 +3,40 @@ package swarm // import "github.com/docker/docker/api/server/router/swarm"
...
+	GetService(idOrName string, insertDefaults bool) (swarm.Service, error)
+	CreateService(swarm.ServiceSpec, string, bool) (*types.ServiceCreateResponse, error)
+	UpdateService(string, uint64, swarm.ServiceSpec, types.ServiceUpdateOptions, bool) (*swarm.ServiceUpdateResponse, error)
 	RemoveService(string) error
-	ServiceLogs(context.Context, *backend.LogSelector, *basictypes.ContainerLogsOptions) (<-chan *backend.LogMessage, error)
-	GetNodes(basictypes.NodeListOptions) ([]types.Node, error)
-	GetNode(string) (types.Node, error)
-	UpdateNode(string, uint64, types.NodeSpec) error
+	ServiceLogs(context.Context, *backend.LogSelector, *types.ContainerLogsOptions) (<-chan *backend.LogMessage, error)
+	GetNodes(types.NodeListOptions) ([]swarm.Node, error)
```
Comment (rumpl) [inline api/server/router/swarm/backend.go:27]: Maybe it's in another commit, shouldn't `types.NodeListOptions` also go to `api/types/Swarm`?

**Review by rumpl — APPROVED** [review]
I have one question but LGTM otherwise

File: api/server/router/swarm/backend.go (hunk 1 above)
Comment (thaJeztah) [reply api/server/router/swarm/backend.go:27]: Think I didn't get to those (nodes, tasks) yet. I think I was still somewhat looking if everything should be _swarm_ or if we should consider some of those to be separate entities (service, node, task), but doing so may require some additional thinking.

**Pull request merged**